Water temperature fix

SpookyTheFox opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Stop me if you've heard this one before.
The solution freezes up into ice! x4
The solution freezes up into ice!
The solution freezes up into ice! x3
The solution freezes up into ice!

You get the idea.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

Currently, if you dispense water into a container from a soda dispenser, like the one in the bar, it comes out at T0C + 0.85, which works out to 274K. If you look over in code\modules\reagents\chemistry\recipes\others.dm , you can see with /datum/chemical_reaction/ice causes water to turn to ice at this temperature. At 275K, ice turns into water. To prevent the soda dispenser from spamming the chemical reaction message and slowly freezing dispensed water into ice, I have changed it to T0C + 1.35, which works out to 274.5K, and causes neither reaction to happen. I already made a similar fix to the service borg shaker, but fixing this one was outside the scope of that PR from a while back.

Fair warning, I do not know why the old value was chosen, it may have another special purpose, but I have not yet encountered it.

## Why It's Good For The Game
This is a fix for something that spams chat and causes unintentional behavior of a chemical reaction.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: tweaked soda dispenser reagent temperatures
/:cl:

The GBP isn't really important to me, this is a bug on a TG downstream and so I am offering my very simple fix up here for the benefit of all spacemans.

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SpookyTheFox committed:
very slight tweak

"The solution freezes up into ice! x542"
BE GONE, THOT!

SpookyTheFox committed:
comment

commented the change

maxymax13 commented:
this is like the second time a pr like this exists. the better solution is to remove the ice reagent so i dont have to hear BLRRRBRRBRBBWWB every 5 seconds ever again

ExcessiveUseOfCobblestone reviewed (changes requested):
Make a define `#define WATER_MATTERSTATE_CHANGE_TEMP 245.5` or whatever

your change here would be `dispensed_temperature = WATER_MATTERSTATE_CHANGE_TEMP`

https://github.com/tgstation/tgstation/blob/620a158e49a3aeb5be8a8141ba8039374976a2e1/code/modules/reagents/chemistry/recipes/others.dm#L788

`required_temp = WATER_MATTERSTATE_CHANGE_TEMP-0.5`

https://github.com/tgstation/tgstation/blob/620a158e49a3aeb5be8a8141ba8039374976a2e1/code/modules/reagents/chemistry/recipes/others.dm#L804

`required_temp = WATER_MATTERSTATE_CHANGE_TEMP+0.5`

never worry about this dumb bug ever again

SpookyTheFox committed:
Merge branch 'tgstation:master' into iceMeltFix

SpookyTheFox committed:
Update reagents.dm

Created a define for the melting/freezing point of water

SpookyTheFox committed:
Update others.dm

changed 274 and 275 limits for water to WATER_MATTERSTATE_CHANGE_TEMP +-0.5

SpookyTheFox committed:
defined drink dispenser temperature

SpookyTheFox committed:
Added dispense temp define to service borg shaker

Sets the drink shaker reagents all to 274.5 with a define for future use. There is a 1 degree dead band for the water/ice reaction here, doing this makes it so water stays water and ice stays ice while still inside the shaker.

SpookyTheFox committed:
Made the beer cold too

Did you know you've been drinking 26C/80F drinks this entire time? Gross.
More importantly, this gets the reagents all the same temperature, so that mixing drinks with the much colder stuff from the soda dispenser doesn't immediately raise the temperature and start things like the ice melt reaction.

SpookyTheFox committed:
forgot a define in here

I put the define on the emagged borg shaker too, so that any drinks mixed in will not be an off temperature and start the ice melt reaction.

SpookyTheFox commented:
I got the bar drink dispenser too, so mixed drinks containing ice don't start the reaction either. You've all been drinking 80F/26C warm, gross drinks this entire time. Also downloaded, compiled, and manually play tested a bunch of stuff to make sure it's all working now.

SpookyTheFox reviewed (commented):
Done!

ExcessiveUseOfCobblestone commented on code/__DEFINES/reagents.dm:
```diff
@@ -175,3 +175,5 @@
 #define BLASTOFF_DANCE_MOVE_CHANCE_PER_UNIT 3
 #define BLASTOFF_DANCE_MOVES_PER_SUPER_MOVE 3
+
+#define WATER_MATTERSTATE_CHANGE_TEMP 245.5 //This is the center of a 1 degree deadband in which water will neither freeze to ice nor melt to liquid
```
```suggestion
///This is the center of a 1 degree deadband in which water will neither freeze to ice nor melt to liquid
#define WATER_MATTERSTATE_CHANGE_TEMP 245.5 
```

ExcessiveUseOfCobblestone committed:
Update code/__DEFINES/reagents.dm
